### ISSUE TITLE
Workers sometimes get confused at room exits

### ISSUE THREAD

**Issue by ryagas** [ScreepsOCS/screeps.behaviour-action-pattern#1085]
Replay of issue occurring:

screeps.com/s/JaQyPJ

Workers bounce in and out of the north exit in the replay.

Relevant log:

`[5:39:47 PM] 18304705 Action isValidAction {"actionName":"bulldozing","behaviourName":"worker","creepName":"worker-400-7","valid":true,"Action":"isValidAction"}
[5:39:47 PM] 18304705 Action isAddableAction {"actionName":"bulldozing","behaviourName":"worker","creepName":"worker-400-7","addable":true,"Action":"isAddableAction"}
[5:39:47 PM] 18304705 Action isValidAction {"actionName":"repairing","behaviourName":"worker","creepName":"worker-400-7","valid":true,"Action":"isValidAction"}
[5:39:47 PM] 18304705 Action isAddableAction {"actionName":"repairing","behaviourName":"worker","creepName":"worker-400-7","addable":true,"Action":"isAddableAction"}
[5:39:47 PM] 18304705 Action isValidAction {"actionName":"feeding","behaviourName":"worker","creepName":"worker-400-7","valid":false,"Action":"isValidAction"}
[5:39:47 PM] 18304705 Action isValidAction {"actionName":"building","behaviourName":"worker","creepName":"worker-400-7","valid":true,"Action":"isValidAction"}
[5:39:47 PM] 18304705 Action isAddableAction {"actionName":"building","behaviourName":"worker","creepName":"worker-400-7","addable":true,"Action":"isAddableAction"}
[5:39:47 PM] 18304705 Action building {"creepName":"worker-400-7","assign":"building","target":"58d96cad9f9ea168314c7e1c","Action":"assign"}`

@karlthepagan hazarded a guess:
> So they think their assigned action should be building, but that makes them path out of the room which might clear the assignment?

**Comment by ryagas** [comment]
Apparently, the worker in the log (as well as others) was trying to path to the upper right corner of the same room as the replay. The location was 46,1, where a road construction site was. They went on to other things once the sites were finished.

**Comment by chobobobo** [comment]
Yeah, this is related to https://github.com/ScreepsOCS/screeps.behaviour-action-pattern/issues/996 I fixed it for remoteWorkers some time back, and that fix should be generalizable, let me look it up
